builder/dockerfile/evaluator.go

```diff
@@ -192,12 +111,30 @@ type dispatchState struct {
 	imageID    string
 	baseImage  builder.Image
 	stageName  string
+	buildArgs  *buildArgs
 }
 
-func newDispatchState() *dispatchState {
-	return &dispatchState{runConfig: &container.Config{}}
+type stageBuild struct {
+	state   *dispatchState
+	shlex   *ShellLex
+	builder *Builder
+	source  builder.Source
+}
```

Comment:
Ok, I found the "stageBuild" name more explicit (it represents the whole state and dependencies required to build a stage), but I agree it would simplify the diff. Might be worth adding a Naming PR after that though